main/src/ca/uwaterloo/flix/language/phase/Dependencies.scala

```diff
@@ -0,0 +1,88 @@
+package ca.uwaterloo.flix.language.phase
+
+import ca.uwaterloo.flix.api.Flix
+import ca.uwaterloo.flix.api.lsp.{Consumer, Visitor}
+import ca.uwaterloo.flix.api.lsp.acceptors.AllAcceptor
+import ca.uwaterloo.flix.language.ast.{SourceLocation, Type, TypeConstructor, TypedAst}
+import ca.uwaterloo.flix.language.ast.TypedAst.Root
+import ca.uwaterloo.flix.language.ast.shared.{DependencyGraph, Input, SymUse}
+import ca.uwaterloo.flix.language.dbg.AstPrinter.*
+import ca.uwaterloo.flix.util.collection.MultiMap
+
+object Dependencies {
+
+  /** Checks the safety and well-formedness of `root`. */
+  def run(root: Root)(implicit flix: Flix): (Root, Unit) = flix.phaseNew("Dependencies") {
```

Comment:
Is returning Unit instead of List standard for infallible phases?

Reply:
Yes

Reply:
I don't like it but ok